clang/include/clang/Basic/LangOptions.def

```diff
@@ -273,6 +273,7 @@ LANGOPT(
     "get/operator[], get_id/operator[] and get_global_id/get_global_linear_id "
     "in SYCL class id, iterm and nd_iterm")
 LANGOPT(SYCLDisableRangeRounding, 1, 0, "Disable parallel for range rounding")
+LANGOPT(SYCLEnableIntHeader, 1, 0, "Enable diagnostics based on SYCL integration header")
```

Comment:
As a nit, I'd probably prefer these be `SYCLEEnableIntHeaderDiags` (or something to that effect, and in addition, on every identifier below), since this looks like it is enabling the integration header, not the diags.

Reply:
This help text doesn't really tell users what this option does or why they'd want to change it (not that users should be setting cc1 options they don't understand). A more direct way to say it is: "Enable SYCL integration header-related diagnostics" but that begs the question of "why is it valid to let the user disable this?".

Unfortunately, I don't know a better way to phrase the help text, so it's probably fine as-is.

FWIW, I was hoping we wouldn't need to introduce a new cc1 option for this -- I was hoping we had existing cc1 options that we could check instead. Did we validate that we need the additional option?

Reply:
> FWIW, I was hoping we wouldn't need to introduce a new cc1 option for this -- I was hoping we had existing cc1 options that we could check instead. Did we validate that we need the additional option?

Unfortunately, I don't think that we can somehow understand that integration header was included yet. For FE it looks like a normal piece of code included via "-include" option, so there was no additional sycl options added for this purpose. The approach implemented here was described by @AGindinson in the offline email thread.
If you have some ideas how to avoid adding a new option and reach the same behavior I'm fully open for any suggestions.

Reply:
Ah, that we didn't add an `-include-sycl-integration-header` option is biting us. We may want to revisit that decision some day, but today is not that day. :-)